network/p2p/middleware.go

```diff
@@ -409,7 +410,13 @@ func (m *Middleware) processMessage(msg *message.Message) {
 func (m *Middleware) Publish(msg *message.Message, channel network.Channel) error {
 
 	// convert the message to bytes to be put on the wire.
-	data, err := msg.Marshal()
+	var data []byte
+	var err error
+	bs := binstat.EnterTime("~3net:wire<4message2protobuf")
```

Comment:
@yhassanzadeh13 thanks for the comment! I have now changed the root of the string to a constant because the roots are used often. It's also easy to grep the constants and this is the recommended way to use the new `binstat` tool to uncomment / unhide instrumentation for a particular bin root.